lib/service/service.go

```diff
@@ -3239,6 +3247,111 @@ func (process *TeleportProcess) initDiagnosticService() error {
 	return nil
 }
 
+// initDebugService starts debug service serving endpoints used for
+// troubleshooting the instance.
+func (process *TeleportProcess) initDebugService() error {
+	logger := process.logger.With(teleport.ComponentKey, teleport.Component(teleport.ComponentDebug, process.id))
+
+	// logPProfMiddleware logs /debug/pprof requests.
+	logPProfMiddleware := func(next http.HandlerFunc) http.HandlerFunc {
+		return func(w http.ResponseWriter, r *http.Request) {
+			seconds := r.URL.Query().Get("seconds")
+			if seconds == "" {
+				seconds = "default"
+			}
+
+			logger.InfoContext(
+				process.ExitContext(),
+				"Collecting pprof profile.",
+				"profile", strings.TrimSuffix(r.URL.Path, constants.PProfEndpointsPrefix),
+				"seconds", seconds,
+			)
+		}
+	}
+
+	mux := http.NewServeMux()
+	mux.HandleFunc(constants.PProfEndpointsPrefix, logPProfMiddleware(pprof.Index))
+	mux.HandleFunc(constants.PProfEndpointsPrefix+"cmdline", logPProfMiddleware(pprof.Cmdline))
+	mux.HandleFunc(constants.PProfEndpointsPrefix+"profile", logPProfMiddleware(pprof.Profile))
+	mux.HandleFunc(constants.PProfEndpointsPrefix+"symbol", logPProfMiddleware(pprof.Symbol))
+	mux.HandleFunc(constants.PProfEndpointsPrefix+"trace", logPProfMiddleware(pprof.Trace))
+
+	mux.HandleFunc(constants.DebugServiceLogLevelEndpoint, func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == constants.DebugServiceGetLogLevelMethod {
+			w.Write([]byte(process.Config.LoggerLevel.Level().String()))
+			return
+		}
+
+		if r.Method != constants.DebugServiceSetLogLevelMethod {
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		// The largest supported log level string representation is 5 characters.
+		// We don't support setting custom levels such as DEBUG+1.
+		rawLevel := make([]byte, 5)
+		defer r.Body.Close()
+		if _, err := r.Body.Read(rawLevel); err != nil && errors.Is(err, io.EOF) {
+			w.WriteHeader(http.StatusUnprocessableEntity)
+			w.Write([]byte("Unable to read request body."))
+			return
+		}
+
+		var level slog.Level
+		if err := level.UnmarshalText(rawLevel); err != nil {
```

Comment:
Wouldn't this fail for `INFO` or `WARN` since they are supposed to be 4 bytes but `rawLevel` has 5?

What's the reason we are reading 5 bytes? Is it because we don't want to support `+-`?

Reply:
How is `TRACE` being handled?

Reply:
I would have just done something like 

```
level := io.ReadAll(io.LimitedReader(resp.Body, 1024))
```

Reply:
Updated to use `io.LimitedReader`. I also added tests to cover setting all the supported log levels.